source/common/grpc/async_client_impl.cc

```diff
@@ -226,10 +228,14 @@ AsyncRequestImpl::AsyncRequestImpl(AsyncClientImpl& parent, absl::string_view se
     : AsyncStreamImpl(parent, service_full_name, method_name, *this, options),
       request_(std::move(request)), callbacks_(callbacks) {
 
-  current_span_ = parent_span.spawnChild(Tracing::EgressConfig::get(),
-                                         "async " + parent.remote_cluster_name_ + " egress",
-                                         parent.time_source_.systemTime());
+  current_span_ =
+      parent_span.spawnChild(Tracing::EgressConfig::get(),
+                             absl::StrCat("async ", service_full_name, ".", method_name, " egress"),
+                             parent.time_source_.systemTime());
   current_span_->setTag(Tracing::Tags::get().UpstreamCluster, parent.remote_cluster_name_);
+  current_span_->setTag(Tracing::Tags::get().UpstreamAddress, parent.host_name_.empty()
+                                                                  ? parent.remote_cluster_name_
```

Comment:
It looks odd to mix up cluster names and host names in one field. Maybe move the empty check into `if` outside?

Reply:
This logic is due to the gRPC config proto

https://github.com/envoyproxy/envoy/blob/53f95570a7e9d7fa5414a4bc145852524fa017ed/api/envoy/config/core/v3/grpc_service.proto#L40

> If this field is not set, the authority header value will be `cluster_name`.

So when `host_name_` is empty, the upstream cluster address is `remote_cluster_name_`. That is what this tag captures.